ratelimit/backends/redis.py

```diff
@@ -1,22 +1,21 @@
-import time
-
 from aredis import StrictRedis
 from aredis.pipeline import StrictPipeline, WatchError
 
 from ..rule import Rule, RULENAMES
 from . import BaseBackend
 
+
 DECREASE_SCRIPT = """
-for i, key in ipairs(KEYS) do
-    local value = tonumber(redis.call('GET', key))
-    if not value or value < 1 then
-        return false
+for i = 1, #KEYS do
+    local value = redis.pcall('GET', KEYS[i])
+    if value and tonumber(value) < 1 then
+        return 0
     end
 end
-for i, key in ipairs(KEYS) do
-    redis.call('DECR', key)
+for i, key in pairs(KEYS) do
+    redis.pcall('DECR', key)
 end
-return true
+return 1
```

Comment:
lgtm, using 1/o instead of true will avoid redis/lua conversion issues it seems
pcall I'm not sure, is it to avoid a potential error being trapped ?

Reply:
I think you are right, it should be changed to call.